inc/namespace.php

```diff
@@ -156,7 +157,17 @@ function add_docs_for_group( Group $group, string $doc_dir ) : Group {
 		}
 
 		$file = $leaf->getRealPath();
+
 		$doc = parse_file( $file, $doc_dir );
+
+		// If the `group-title` meta is set in the README file, change the group's title.
+		if ( strtolower( $leaf->getBasename() ) === 'readme.md' ) {
+			$group_title = $doc->get_meta( 'group-title' );
```

Comment:
The `parse_file` function sets the `title` meta data from the first h1 heading in the content if it exists already, so rather than introduce a new meta value we can use the existing `title` meta value here instead. Fine to also support a new more specific meta data value but it should check the title meta and set the group value from that first if available.

Reply:
The `title` that `parse_file()` finds is used for the `<title>` tag  in the html header. That was added for SEO purposes with the deliberate intention of being able to make it different from the `h1` on the page.

This group title is for the left hand navigation that is automatically built. There are a lot of reasons you don't want it to be the same as either the html header title or the h1 title on the page. Not least of which is brevity because of space constraints. 

Currently the group title (for the left hand navigation) is set either manually (for the three groups in `other-docs`) or during the call `register_module()`. So it is not currently related to the md files in the docs folder at all. In fact we already have one example of the module/group title being different from the h1 in the docs (module title `SSO`, docs h1 "Single Sign On"). Falling back to the h1 would change the existing documentation navigation.

Reply:
> The title that parse_file() finds is used for the <title> tag in the html header. That was added for SEO purposes with the deliberate intention of being able to make it different from the h1 on the page.

The docs rendered by this code are only shown in the admin so SEO shouldn't be a factor. The titles in the menu of all the group subpages are being pulled from the h1 tags though.

This is what I'm referring to where it's reading the h1 for the default title: 
https://github.com/humanmade/altis-documentation/blob/master/inc/namespace.php#L268-L271

Regarding the `<title>` tag are you referring to the dev docs site?

> Falling back to the h1 would change the existing documentation navigation.

I think that's ok as it will only change it for the group headings, and it will make it consistent with https://docs.altis-dxp.com/

Maybe we could add the option to specify a `menu-title` meta value instead but I think that's out of scope here.